build/README.md

```diff
@@ -0,0 +1,164 @@
+# Building fms-hf-tuning as an Image
+
+The Dockerfile provides a way of running fms-hf-tuning SFT Trainer. It installs the dependencies needed and adds two additional scripts that helps to parse arguments to pass to SFT Trainer. The `accelerate_launch.py` script is run by default when running the image to trigger SFT trainer for single or multi GPU by parsing arguments and running `accelerate launch launch_training.py`. 
+
+## Configuration
+
+The scripts accept a JSON formatted config which are set by environment variables. `SFT_TRAINER_CONFIG_JSON_PATH` can be set to the mounted path of the JSON config. Alternatively, `SFT_TRAINER_CONFIG_JSON_ENV_VAR` can be set to the encoded JSON config using the below function:
+
+```py
+import base64
+
+ def encode_json(my_json_string):
+    base64_bytes = base64.b64encode(my_json_string.encode("ascii"))
+    txt = base64_bytes.decode("ascii")
+    return txt
+```
+
+The keys for the JSON config are all of the flags available to use with [SFT Trainer](https://huggingface.co/docs/trl/sft_trainer#trl.SFTTrainer).
+
+For configuring `accelerate launch`, use key `multiGPU` and pass the set of flags accepted by [accelerate launch](https://huggingface.co/docs/accelerate/package_reference/cli#accelerate-launch).
+
+For example, the below config is used for running with two GPUs and FSDP for PEFT tuning:
+
+Note that `num_processes` which is the total number of processes to be launched in parallel,should match the number of GPUs to run on.
+
+```json
+{
+    "multiGPU": {
+        "num_machines": 1,
+        "main_process_port": 1234,
+        "num_processes": 2,
+        "use_fsdp": true,
+        "fsdp_backward_prefetch_policy": "TRANSFORMER_BASED_WRAP",
+        "fsdp_sharding_strategy": 1,
+        "fsdp_state_dict_type": "FULL_STATE_DICT",
+        "fsdp_cpu_ram_efficient_loading": true,
+        "fsdp_sync_module_states": true
```

Comment:
Trying to think out loud, should we call out that these accelerate parameters are flattened like typically provided to accelerate cli? (I see CLI arg parser is used in the code https://github.com/foundation-model-stack/fms-hf-tuning/pull/98/files#diff-3918b6c0fe504e55b66f13d0cbc2958364e228b0bd7244c67efc5732c5f067f5R91-R92). The reason being users might get confused with this being in JSON to the typical accelerate YAML config (https://github.com/foundation-model-stack/fms-hf-tuning/blob/4cc09cbe4cedff71ef2260ce4847de836df45c90/fixtures/accelerate_fsdp_defaults.yaml)  Where arguments are not flattened and can be a dict type

https://github.com/foundation-model-stack/fms-hf-tuning/blob/4cc09cbe4cedff71ef2260ce4847de836df45c90/fixtures/accelerate_fsdp_defaults.yaml#L11-L15

Reply:
https://github.com/foundation-model-stack/fms-hf-tuning/pull/98/files#diff-4b71ec736aee936feb417e5193e800a1014a8e2df2a5b5b92fd78fbd3d781dbdR61 

May be this point is calling out that difference.

Reply:
@kmehant I see, I think maybe just put a link to the [accelerate docs](https://huggingface.co/docs/accelerate/en/package_reference/cli#accelerate-launch), it should be clear what command line arguments are available. When we talk about command line args, it should be unambiguous that it is a flattened dict

I think if by @kmehant 's suggestion, simply replacing `multGPU` with `accelerate_launch_args` it would make it even clearer.